### PR TITLE
userdbctl: show UID ranges in output too

### PR DESCRIPTION
**Pull request by poettering**
This tweaks output of "userdbctl" user/group listing output, and will show UID boundaries (i.e. system user boundary, and similar) next to the users actually defined.

it also adds a splash of color and unicode glyphs to differentiate the output fields.

### CODE REVIEW

File: src/basic/glyph-util.c [hunk 1]
```diff
@@ -89,9 +92,11 @@ const char *special_glyph(SpecialGlyph code) {
                         [SPECIAL_GLYPH_LIGHT_SHADE]             = "\342\226\221",             /* ░ */
                         [SPECIAL_GLYPH_DARK_SHADE]              = "\342\226\223",             /* ▒ */
                         [SPECIAL_GLYPH_SIGMA]                   = "\316\243",                 /* Σ */
+                        [SPECIAL_GLYPH_ARROW_UP]                = "\u2191",                   /* ↑ (actually called: UPWARDS ARROW) */
+                        [SPECIAL_GLYPH_ARROW_DOWN]              = "\u2193",                   /* ↑ (actually called: DOWNWARDS ARROW) */
```
Comment (yuwata) [inline src/basic/glyph-util.c]: Typo in comment.

File: src/basic/glyph-util.c [hunk 2]
```diff
@@ -75,6 +77,7 @@ const char *special_glyph(SpecialGlyph code) {
                         [SPECIAL_GLYPH_TREE_BRANCH]             = "\342\224\234\342\224\200", /* ├─ */
                         [SPECIAL_GLYPH_TREE_RIGHT]              = "\342\224\224\342\224\200", /* └─ */
                         [SPECIAL_GLYPH_TREE_SPACE]              = "  ",                       /*    */
+                        [SPECIAL_GLYPH_TREE_TOP]                = "\u250C\342\224\200",       /* ┌─ */
```
Comment (yuwata) [inline src/basic/glyph-util.c:81]: ASCII version is missing.

File: src/userdb/userdbctl.c [hunk 3]
```diff
@@ -194,7 +310,7 @@ static int display_user(int argc, char *argv[], void *userdata) {
 
                 if (arg_legend) {
                         if (table_get_rows(table) > 1)
-                                printf("\n%zu users listed.\n", table_get_rows(table) - 1);
+                                printf("\n%zu users listed.\n", table_get_rows(table) - 1 - boundary_lines);
```
Comment (yuwata) [inline src/userdb/userdbctl.c:341]: I guess `- boundary_lines * 2`?

File: src/userdb/userdbctl.c [hunk 4]
```diff
@@ -359,7 +545,7 @@ static int display_group(int argc, char *argv[], void *userdata) {
 
                 if (arg_legend) {
                         if (table_get_rows(table) > 1)
-                                printf("\n%zu groups listed.\n", table_get_rows(table) - 1);
+                                printf("\n%zu groups listed.\n", table_get_rows(table) - 1 - boundary_lines);
```
Comment (yuwata) [inline src/userdb/userdbctl.c:581]: Also here.

File: src/userdb/userdbctl.c (hunk 3 above)
Comment (poettering) [reply src/userdb/userdbctl.c:341]: ouch, indeed

**Comment by poettering** [comment]
Fixed. (Fixed it in the return value, i.e. in the producer, not the consumer of the value)

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**